src/Unity.Mathematics.Editor/PrimitiveVectorDrawer.cs

```diff
@@ -42,7 +42,15 @@ public override void OnGUI(Rect position, SerializedProperty property, GUIConten
         {
             var subLabels = Content.labels4;
             var startIter = "x";
-            switch (property.type[property.type.Length - 1])
+            var propertyType = property.type;
+            var isManagedRef = property.type.StartsWith("managedReference", StringComparison.Ordinal);
+            if (isManagedRef)
+            {
+                var startIndex = "managedReference<".Length;
+                var length = propertyType.Length - startIndex - 1;
+                propertyType = propertyType.Substring("managedReference<".Length, length);
+            }
+            switch (propertyType[propertyType.Length - 1])
```

Comment:
`SubString()` will allocate a string, which we definitely don't want inside the OnGUI loop. I'd suggest doing this check one time and caching the result on the drawer instance. A single drawer instance should not be used to draw more than one type, so it should be safe to do

Reply:
Done in b0d4bfb